packages/eslint-plugin/src/util/explicitReturnTypeUtils.ts

```diff
@@ -37,6 +37,51 @@ function isPropertyDefinitionWithTypeAnnotation(
   );
 }
 
+/**
+ * Checks if a node belongs to:
+ * ```
+ * foo(() => 1)
+ * ```
+ */
+function isFunctionArgument(
+  parent: TSESTree.Node,
+  callee?: FunctionExpression,
+): parent is TSESTree.CallExpression {
+  return (
+    parent.type === AST_NODE_TYPES.CallExpression &&
+    // make sure this isn't an IIFE
+    parent.callee !== callee
+  );
+}
+
+/**
+ * Checks if a node is an expression in a JSX attribute assignment
+ * ```
+ * <Comp x={...} />
+ * ```
+ */
+function isJSXAttribute(
+  node: TSESTree.Node,
+): node is TSESTree.JSXExpressionContainer | TSESTree.JSXSpreadAttribute {
+  return (
+    node.type === AST_NODE_TYPES.JSXExpressionContainer ||
```

Comment:
This is technically the RHS of a `JSXAttribute`, so this check could also recurse one level higher if that is preferred. This should be the only way to have a function supplied in JSX though.